runtime/etc/Makefile.mli-shared

```diff
@@ -40,6 +43,9 @@ buildClient: $(TMPBINNAME)
 # need to pass on the $(GEN_CFLAGS) variable, because we're building a shared
 # object and $(GEN_CFLAGS) contains the -fPIC flag.
 #
+# Always run this, even when compiling with `--llvm`. TODO: Verify that we
+# are building with CHPL_TARGET_COMPILER here.
```

Comment:
Is this TODO something for this PR?

Reply:
Agreed -- it looks like a TODO for this PR, that just hasn't been removed. More broadly, is this "Always run" paragraph really necessary, i.e. is there something that would make a reader think there are circumstances in which this might not be run? If so maybe we should add more explanation about why we're doing it.

Reply:
By verify I mean via assert, or whatever the equivalent is. Instead of thinking it's right have something in the Makefiles or build system verify it to be true. I can change the word "Verify" to "Assert" but I don't know if that's possible to do in Makefiles.

Reply:
I added two different TODOs.